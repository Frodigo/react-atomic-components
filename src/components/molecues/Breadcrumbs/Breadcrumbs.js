import React from 'react';
import PropTypes from 'prop-types';

import classes from './Breadcrumbs.module.css';

const Breadcrumbs = (props) => {
    const { items } = props;
    const { ...rest } = props;

    const breadcrumbsItems = [...items];

    const activeItem = breadcrumbsItems.pop();

    const shouldDisplayBreadcrumbsItems = breadcrumbsItems
        ? breadcrumbsItems.map((item) => {
              return (
                  <a
                      href={item.url}
                      className={classes.breadcrumbsItem}
                      key={item.label}
                  >
                      <span className={classes.label}>{item.label}</span>
                  </a>
              );
          })
        : null;

    const shouldDisplayActiveItem = activeItem ? (
        <span className={classes.activeItem}>{activeItem.label}</span>
    ) : null;

    return (
        <nav className={classes.breadcrumbs} {...rest}>
            <div className={classes.container}>
                {shouldDisplayBreadcrumbsItems}
                {shouldDisplayActiveItem}
            </div>
        </nav>
    );
};

Breadcrumbs.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            label: PropTypes.string.isRequired,
            url: PropTypes.string.isRequired
        })
    ).isRequired
};

export default Breadcrumbs;
