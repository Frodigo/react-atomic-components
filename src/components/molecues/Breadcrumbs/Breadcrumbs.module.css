.breadcrumbs {
    border-bottom: solid 1px var(--color-gray-dark-2);
    padding: var(--spacing-md);
}

.container {
    align-items: center;
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
}

.breadcrumbsItem {
    color: var(--color-primary);
    font-weight: normal;
    line-height: 1.43;
    text-decoration: none;
}

.breadcrumbsItem::after {
    content: '/';
    margin: 0 3px;
    text-decoration: none;
}

.label {
    text-decoration: underline;
}

.activeItem {
    composes: breadcrumbsItem;
    color: var(--color-black);
    text-decoration: none;
}

.activeItem::after {
    display: none;
}

@media (min-width: 1300px) {
    .container {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
    }
}
