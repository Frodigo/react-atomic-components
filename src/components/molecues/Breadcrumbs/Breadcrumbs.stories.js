import React from 'react';

import { default as BreadcrumbsComponent } from './Breadcrumbs';

export default {
    title: 'Molecues/Breadcrumbs',
    component: BreadcrumbsComponent
};

const Template = (args) => <BreadcrumbsComponent {...args} />;

export const Default = Template.bind({});
Default.args = {
    items: [
        {
            label: 'Home',
            url: '#'
        },
        {
            label: 'Blog',
            url: '#'
        },
        {
            label: 'Lorem ipsum',
            url: '#'
        }
    ]
};
