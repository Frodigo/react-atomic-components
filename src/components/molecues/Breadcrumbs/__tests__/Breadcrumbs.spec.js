import React from 'react';
import { render } from '@testing-library/react';

import Breadcrumbs from '../Breadcrumbs';

describe('Breadcrumbs should be rendered', () => {
    const mockedItems = [
        {
            label: 'Home',
            url: '#'
        },
        {
            label: 'Blog',
            url: '#'
        },
        {
            label: 'Lorem ipsum',
            url: '#'
        }
    ];

    test('Breadcrumbs renders correctly', () => {
        const { container } = render(<Breadcrumbs items={mockedItems} />);

        expect(container).toMatchSnapshot();
    });
});
