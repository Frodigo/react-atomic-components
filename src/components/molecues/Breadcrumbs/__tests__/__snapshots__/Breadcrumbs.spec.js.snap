// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should be rendered Breadcrumbs renders correctly 1`] = `
<div>
  <nav
    class="breadcrumbs"
    items="[object Object],[object Object],[object Object]"
  >
    <div
      class="container"
    >
      <a
        class="breadcrumbsItem"
        href="#"
      >
        <span
          class="label"
        >
          Home
        </span>
      </a>
      <a
        class="breadcrumbsItem"
        href="#"
      >
        <span
          class="label"
        >
          Blog
        </span>
      </a>
      <span
        class="activeItem"
      >
        Lorem ipsum
      </span>
    </div>
  </nav>
</div>
`;
